packages/core/src/components/Table/TableVirtualizedBody/TableVirtualizedBody.tsx

```diff
@@ -61,30 +74,42 @@ const TableVirtualizedBody = forwardRef(
         scrollOffset: number;
         scrollUpdateWasRequested: boolean;
       }) => {
+        if (virtualizedWithHeader) return;
         onScroll?.(scrollDirection, scrollOffset, scrollUpdateWasRequested);
       },
-      [onScroll]
+      [onScroll, virtualizedWithHeader]
     );
 
     const itemRenderer = useCallback<ComponentType<ListChildComponentProps<TableVirtualizedRow>>>(
       ({ index, style: { width: _width, ...style } }) => {
-        const currentItem = items[index];
+        if (virtualizedWithHeader && index === 0) {
+          return null; //placeholder for virtualized with header
+        }
+        const currentIndex = virtualizedWithHeader ? index - 1 : index;
+        const currentItem = items[currentIndex];
         const element = rowRenderer(currentItem);
+
         return React.cloneElement(element, {
           style: { ...style, ...element.props?.style },
           key: index
         });
       },
-      [items, rowRenderer]
+      [items, rowRenderer, virtualizedWithHeader]
     );
 
     useEffect(() => {
-      markTableAsVirtualized();
-    }, [markTableAsVirtualized]);
+      if (!virtualizedWithHeader) markTableAsVirtualized();
```

Comment:
why does it matter though?

Reply:
Now that we don't need the sync for the header on the scroll it should act the same as non virtualized table, so there is no need to mark it as virtualized